### ISSUE TITLE
Moonstone didn't activate its Reaction ability when discarding down to 8 cards in hand

### ISSUE THREAD

**Issue by mniksa3** [halogenandtoast/ArkhamHorror#460]
**Game ID**: c5091d91-b062-4beb-a2e3-cb654bb37315

**Bug**: When discarded, Moonstone should give me the opportunity to play itself. When discarding from 10 cards in hand down to 8 during the Upkeep phase, I chose to discard Moonstone (as well as Moonlight Ritual) but did not receive a window to play Moonstone. It simply landed in my discard pile.

**Comment by ChooJeremy** [comment]
I have the same issue as well. Moonstone triggers are inconsistent and this is more annoying to play with because debug mode doesn't allow an easy way of fixing the issue, as opposed to other bugs.

**Game ID**: 5dda9433-c57d-4a44-8da2-48b9d46c8521  
In this game, at the scenario Beyond the Gates of Sleep, I discarded my whole hand with Patrice (including the dreamstone) and it did not trigger). I was able to fix this issue by readying the violin (which I already had in play) and using the violin to discard the Moonstone, which caused the reaction icon to appear below the discard pile. With this, I was able to play the moonstone then deduct the extra resource I received from the voilin.

**Game ID**: 56071779-0cb2-4d4d-a324-87f1ecfefd79
In this game, the Moonstone once again failed to trigger when I discarded my whole hand with Patrice. I managed to draw the violin next round, and attempted to fix this by drawing the second copy of the moonstone and discarding with the voilin. This time, no reaction icon appears and I'm unable to play the moonstone at all.
